### ISSUE TITLE
Searching repositories by topic

### ISSUE THREAD

**Issue by jugaltheshah** [NerdWalletOSS/shepherd#352]
Hi folks,

I'm having an issue searching for repositories matching a topic while trying to create a migration.

For example, I created the following shepherd.yml:
```
id: 2021-01-04-test-topic-search
title: |
  chore: test topic searching
adapter:
  type: github
  search_query: org:my-org-name topic:my-topic
hooks:
  apply: echo $SHEPHERD_MIGRATION_DIR
```
I then ran `shepherd checkout .` in the migration folder. This comes up with 0 repositories:
```
$ shepherd checkout .
[before-after-hook]: "Hook()" repurposing warning, use "Hook.Collection()". Read more: https://git.io/upgrade-before-after-hook-to-1.4
✔ Loaded 0 repos

Checked out 0 out of 0 repos
```
If I take the exact same `search_query` and search for it on Github.com, it comes up with the correct number of repos (changing the org & topic, of course).

Any ideas what the issue could be? 

P.S. For `GITHUB_TOKEN`, I used a token with which I was able to update topics, so I assume it should be able to read/search for topics also.

**Comment by nwalters512** [comment]
👋 Thanks for pointing this out! This is something we didn't consider when building the GitHub adapter. Much like the GitHub search results page has different sections/tabs for Repositoriess, Code, Commits, Issues, etc., the GitHub API also differentiates between them via different endpoints - `/search/code`, `/search/repositories`, etc. We specifically use the code search endpoint: 

https://github.com/NerdWalletOSS/shepherd/blob/d47f53023e03838a94782508da9037187706d883/src/adapters/github.ts#L74

To give more flexibility, I'd propose a `search_type` config property for the GitHub adapter that would allow the user to pick between the `code` and `repositories` search endpoints. Would you be willing to open a PR adding that?

**Comment by jugaltheshah** [comment]
Aha, you beat me to it, I was just testing search for topics with `octokit.search.code` vs `octokit.search.repos`. Sure, I will open a PR for that.

**Comment by jugaltheshah** [comment]
Still need to fix up tests, but can I get a soft review of https://github.com/NerdWalletOSS/shepherd/pull/353 to make sure it's on the right track @nwalters512 @aorinevo?

**Comment by aorinevo** [comment]
Looks good to me.  Side note, reviewing this PR reminds me that we still have an Octokit upgrade that's very overdue.

**Comment by jugaltheshah** [comment]
Addressed via https://github.com/NerdWalletOSS/shepherd/pull/353 and included in `v1.12.0` and beyond, closing.
